File: 0704_Binary_Search/main.cpp
class Solution {
  public:
    int search(vector<int> &nums, int target) {
        if (nums.empty()) {
            return -1;
        }
        int l = 0, r = nums.size() - 1;
        while (l < r) {
            int mid = (l + r + 1) >> 1;
            if (nums[mid] <= target) {
                l = mid;
            } else {
                r = mid - 1;
            }
        }
        if (nums[l] == target) {
            return l;
        }
        return -1;
    }
};

File: 0704_Binary_Search/golang/main.go
func search(nums []int, target int) int {
	if nums == nil || len(nums) == 0 {
		return -1
	}
	n := len(nums)
	l, r := 0, n-1
	for l < r {
		mid := l + (r-l+1)/2
		if nums[mid] <= target {
			l = mid
		} else {
			r = mid - 1
		}
	}
	if nums[l] == target {
		return l
	}
	return -1
}